relational_operators/UnionAllOperator.cpp

```diff
@@ -0,0 +1,144 @@
+    WorkOrdersContainer *container,
+    QueryContext *query_context,
+    StorageManager *storage_manager,
+    InsertDestination *output_destination,
+    const std::size_t relation_index) {
+  if (input_relations_are_stored_[relation_index]) {
+    const std::vector<block_id> &all_blocks = input_relations_block_ids_.at(relation_index);
+    for (const block_id input_block_id : all_blocks) {
+      container->addNormalWorkOrder(
+          new UnionAllWorkOrder(
+              query_id_,
+              input_relations_[relation_index],
+              input_block_id,
+              select_attribute_ids_[relation_index],
+              output_destination,
+              storage_manager),
+          op_index_);
+    }
+  } else {
+    std::size_t num_generated = num_workorders_generated_[relation_index];
+    const std::vector<block_id> &all_blocks = input_relations_block_ids_[relation_index];
+    while (num_generated < all_blocks .size()) {
+      container->addNormalWorkOrder(
+          new UnionAllWorkOrder(
+              query_id_,
+              input_relations_[relation_index],
+              all_blocks[num_generated],
+              select_attribute_ids_[relation_index],
+              output_destination,
+              storage_manager),
+          op_index_);
+      ++num_generated;
+    }
+    num_workorders_generated_[relation_index] = num_generated;
+  }
+}
+
+bool UnionAllOperator::getAllWorkOrders(
+    WorkOrdersContainer *container,
+    QueryContext *query_context,
+    StorageManager *storage_manager,
+    const tmb::client_id scheduler_client_id,
+    tmb::MessageBus *bus) {
+  DCHECK(query_context != nullptr);
+
+  InsertDestination *output_destination =
+      query_context->getInsertDestination(output_destination_index_);
+
+  if (!stored_generated_) {
+    for (std::size_t relation_index = 0; relation_index < input_relations_.size(); ++relation_index) {
+      if (input_relations_are_stored_[relation_index]) {
+        addWorkOrdersSingleRelation(container,
+                                    query_context,
+                                    storage_manager,
+                                    output_destination,
+                                    relation_index);
+      }
+    }
+    stored_generated_ = true;
+  }
+
+  for (std::size_t relation_index = 0; relation_index < input_relations_.size(); ++relation_index) {
+    if (!input_relations_are_stored_[relation_index]) {
+       addWorkOrdersSingleRelation(container,
+                                   query_context,
+                                   storage_manager,
+                                   output_destination,
+                                   relation_index);
+    }
+  }
+  return stored_generated_ && done_feeding_input_relation_;
+}
+
+bool UnionAllOperator::getAllWorkOrderProtos(WorkOrderProtosContainer* container) {
+  // TODO(tianrun): Add protobuf for UnionAllWorkOrder to support distributed mode.
```

Comment:
Could we create a JIRA ticket for this, or do we expect that there will be a follow-up PR immediately after this one?